internal/provider/resources/code_location.go

```diff
@@ -40,23 +45,70 @@ func (r *CodeLocationResource) Schema(ctx context.Context, req resource.SchemaRe
 
 		Attributes: map[string]schema.Attribute{
 			"name": schema.StringAttribute{
-				MarkdownDescription: "Code Location name",
+				MarkdownDescription: "Code Location name. ",
 				Required:            true,
 			},
 			"image": schema.StringAttribute{
-				MarkdownDescription: "Code Location image",
-				Required:            true,
+				MarkdownDescription: "Code Location image. Git or Image is a required field (mutually exclusive).",
+				Required:            false,
+				Optional:            true,
 			},
 			"code_source": schema.SingleNestedAttribute{
 				Attributes: map[string]schema.Attribute{
+					"module_name": schema.StringAttribute{
+						MarkdownDescription: "Code Location code source module name. One of module_name/package_name/python_file is required (mutually exclusive).",
+						Required:            false,
+						Optional:            true,
+					},
+					"package_name": schema.StringAttribute{
+						MarkdownDescription: "Code Location code source package name. One of module_name/package_name/python_file is required (mutually exclusive).",
+						Required:            false,
+						Optional:            true,
+					},
 					"python_file": schema.StringAttribute{
-						MarkdownDescription: "Code Location code source python file",
-						Required:            true,
+						MarkdownDescription: "Code Location code source python file. One of module_name/package_name/python_file is required (mutually exclusive).",
+						Required:            false,
+						Optional:            true,
 					},
 				},
 				MarkdownDescription: "Code Location code source",
 				Required:            true,
 			},
+			"working_directory": schema.StringAttribute{
+				MarkdownDescription: "Code Location working directory",
+				Required:            false,
+				Optional:            true,
+			},
+			"executable_path": schema.StringAttribute{
+				MarkdownDescription: "Code Location executable path",
```

Comment:
Python? Which executable?

Reply:
https://docs.dagster.io/concepts/code-locations/workspace-files#examples

Reply:
Might be Dagster OSS only? Maybe we don't need to support these attributes

Reply:
is in the dagster cloud graphql api, so we have to support it